### PR TITLE
Initial submission of README file containing local dev setup instruct…

### PR DESCRIPTION
**Pull request by spence-man**
Review request for README & WordPress Setup to merge with development

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,51 @@
...
+  Create the MySQL database.
+
+  ```
+  mysql -u username -p
+  CREATE DATABASE databasename;
+  GRANT ALL PRIVILEGES ON databasename.* TO "wordpressusername"@"hostname" IDENTIFIED BY "password";
+  FLUSH PRIVILEGES;
+  EXIT
+  ```
+  Download WordPress and run the server.
```
Comment (tatwell) [inline README.md]: Let's move the WP core source into this repo. This is what we've done on Blogs:

- https://github.com/chapmanu/blogs/tree/development/public

We've tried it both ways. With our plan to deploy code using Capistrano, having the code in the repo will be simpler. Updates will be done as described there:

- https://github.com/chapmanu/blogs#to-update-the-wp-core

File: README.md [hunk 2]
```diff
@@ -1,2 +1,51 @@
...
+- A consistent local development environment
+- Source control using standard Git workflow practices
+- Modern automated tests
+- Documentation for best practices and standards
+
+## Requirements
+- Homebrew (1.2.3)
+  - MySQL (5.7.18)
+  - Wordpress Command Line (1.2.1)
+- PHP (5.6.30)
```
Comment (tatwell) [inline README.md]: Requirements need to meet those specified here:

- https://wordpress.org/about/requirements/

So PHP 7. Sorry I should have clarified that in the acceptance criteria.

Red Hat (our production/staging servers) uses MariaDB, so try to set that up if possible. As far as I know, it is pretty interchangeable with MySQL (Red Hat aliases mysql to it). Still, I'd like dev/staging/production to be as similar as possible.

**Review by tatwell — CHANGES_REQUESTED** [review]
See comments. I updated Acceptance Criteria in Trello card.

File: README.md (hunk 2 above)
Comment (tatwell) [reply README.md]: For reference, our Ansible Mac workstation playbook installs PHP 7:

https://github.com/chapmanu/ansible-workstation

File: README.md [hunk 3]
```diff
@@ -1,2 +1,66 @@
...
+  FLUSH PRIVILEGES;
+  EXIT
+  ```
+
+  ### Download WordPress and run the server.
+
+  ```
+  cd /public/
+  wp core download
+  php -S localhost:8000
```
Comment (tatwell) [inline README.md]: The public directory (with the WordPress core code) should be in the repository. You'll need to add it.

You'll also want to make sure no sensitive stuff is committed to the repository like config file with passwords. If you have questions about how to handle this, ping me.

 So I'd expect install instructions at this step to be something like:

```
git clone git@github.com:chapmanu/chap-press.git
cd chap-press/public
cp -v wp-config.php{-dist,}

# Update db settings in wp-config.php

php -S localhost:8000
```

**Comment by tatwell** [comment]
@spence-man I succeeded in installing project locally. 👍  (I made some tweaks to instructions.)

I noticed, however, that when I go to the site in the browser, I meet the WordPress install wizard. After install, I'd like to have dev find the site fully installed and running. I'd guess this means making some tweaks to the `wp-config.php-dist` file.

On that point, let's rename `wp-config-sample.php` to `wp-config.php-dist`. That's our conventional way of marking config files.

**Comment by spence-man** [comment]
@tatwell I found a solution for [installing wordpress](http://www.hongkiat.com/blog/wordpress-command-line/) using the wp cli. It creates the config file and bypasses the UI install wizard. Just have to login into wp-admin when it is finished.

**Generates config file**
1.`wp core config --dbname=chappress_dev --dbuser=chappress --dbpass=chappress`
2.`chmod 644 wp-config.php`

**Remove sample file**
3.` rm wp-config-sample.php`

**Create a copy of generated config file**
4. `cp -v wp-config.php wp-config.php-dist`

**Wordpress Install**
5.`wp core install --url=http://localhost:8000/ --title=Chappress-blog --admin_user=chappress-user --admin_password=chappress --admin_email=chappress@gmail.com`

**Login**
6.`http://localhost:8000/wp-admin/`
*without logging in, it doesn't show the admin panel (just the site)

I'm guessing we could eventually put these long terminal entries to a script or automation tool later on. Let me know what you think.

**Comment by tatwell** [comment]
Ugh. All those files make it hard to review PR. I wiped my repo and am starting from scratch with fresh clone. Hit this:

```
$ cp -v wp-config.php{-dist,}
cp: wp-config.php-dist: No such file or directory
```

**Comment by tatwell** [comment]
With `wp-config.php-dist` file, why don't we just pre-fill it with the db settings from the README. Add a comment noting that they are for dev.

In future story, I'd like us to set up a simple way to switch settings based on environment like Rails does. (I think the devs rigged something like that for Blogs.)

**Comment by tatwell** [comment]
Otherwise, it worked like a charm. Nice work.

What about XCode? How does that work? Is that for PHP debugging or is it a Mac thing? It doesn't look like it a requirement. Maybe list it as recommended.

**Comment by tatwell** [comment]
In response to [your comment on installing WordPress](https://github.com/chapmanu/chap-press/pull/1#issuecomment-313475023), I'd recommended putting that info in its own wiki page here in this repository for now. Perhaps link to it in the README.

The info is not essential for devs working on this project, but I like that approach for future WordPress projects so it would be handy to keep for reference.

**Comment by tatwell** [comment]
Excellent. Got site set up locally from scratch smooth and buttery. Only thing missing: I don't see the log. Maybe you can add a log call to the code somewhere so that it will generate the log.

Maybe put it in the config file (assuming it will work there), that way it will be easy to find and can serve as a reference for how to use logging.

**Comment by spence-man** [comment]
Finally realized why the log wasn't showing after I initialized it. The gitignore file was set to ignore all logs (`*.log`). Just made the adjustment.

**Comment by tatwell** [comment]
@spence-man No, you had it right before. Logs should be ignored and not included in repo.

However, when I run the local server and request a page, I'd like to see the log appear at path you noted in README. 

Also, I like the idea of putting a log call at the top or bottom of the config file to make it clear how to use the log in WordPress.

File: public/wp-config.php-dist [hunk 4]
```diff
@@ -92,3 +92,9 @@ if ( !defined('ABSPATH') )
 
 /** Sets up WordPress vars and included files. */
 require_once(ABSPATH . 'wp-settings.php');
+
+/* ChapPress-For logging purposes only, not to be used for live site */
+$messageLog  = '';
+$messageLog .= "[" . date('d-M-Y H:i:s e') . "]" . " - "; 
+$messageLog .= "Hello from the config file \n"; 
+error_log($messageLog, 3, "./wp-content/debug.log");
```
Comment (tatwell) [inline public/wp-config.php-dist:100]: It worked. Nice. I'd prefer a simpler interface but that can be address in the future.

**Review by tatwell — APPROVED** [review]
Looks good. Nice work.

**Pull request merged**